### PR TITLE
Factor out Meta.{Check,Enabled}

### PR DESCRIPTION
**Pull request by jcorbin**
Set's the stage to further redefine levels towards #173.

### CODE REVIEW

File: meta.go [hunk 1]
```diff
@@ -74,3 +74,24 @@ func (m Meta) Clone() Meta {
 	m.Encoder = m.Encoder.Clone()
 	return m
 }
+
+// Enabled returns true if logging a message at a particular level is enabled.
+func (m Meta) Enabled(lvl Level, msg string) bool {
```
Comment (akshayjshah) [inline meta.go]: Why take the message as a param?

File: meta.go [hunk 2]
```diff
@@ -74,3 +74,24 @@ func (m Meta) Clone() Meta {
...
+	case PanicLevel, FatalLevel:
+		// Panic and Fatal should always cause a panic/exit, even if the level
+		// is disabled.
+		break
+	default:
+		if !m.Enabled(lvl, msg) {
+			return nil
+		}
+	}
+	return NewCheckedMessage(log, lvl, msg)
```
Comment (akshayjshah) [inline meta.go]: Unless there's another use case for it, I'm all for un-exporting `NewCheckedMessage`. Existing callers should all have access to a `Meta`, which they can use instead.

File: zwrap/sample_test.go [hunk 3]
```diff
@@ -181,8 +181,7 @@ func TestSamplerCheckPanicFatal(t *testing.T) {
 			}
 		}
 
-		expected := buildExpectation(level, 0, 1, 2, 3, 4)
-		assert.Equal(t, expected, sink.Logs(), "Unexpected output when sampling with Check.")
+		assert.Equal(t, []spy.Log(nil), sink.Logs(), "Unexpected output when sampling with Check.")
```
Comment (akshayjshah) [inline zwrap/sample_test.go]: Yowza...previous behavior was totally busted :(

File: meta.go (hunk 1 above)
Comment (jcorbin) [inline meta.go]: Good qn, I think having the msg only encourages sampling-like things; will drop.

**Pull request merged**